doc/user-guide/push-notifications/Push-notifications.md

```diff
@@ -0,0 +1,217 @@
+node (the most common setup!). Or, for the best performance, you might just skip that remote PubSub
+node altogether. While the whole setup can be incredibly extensible, we see the following
+straightforward uses of it.
+
+### XEP-0357 compliant with local PubSub
+
+ ![](PushNotifications-MIM_with_local_pubsub.png)
+
+This is, historically, the most common setup.
+It allows your clients to enable push notifications via a local PubSub,
+and the IQ stanza is routed internally.
+
+The components that make push notifications possible in MongooseIM in the PubSub-full setup
+add up to the following architecture:
+![](component-architecture-with-PubSub.png)
+
+A direct connection to a push service (e.g. MongoosePush) must be configured on the same MongooseIM
+node, checkout [this][WithMongoosePush] tutorial on how to setup [MongoosePush][].
+
+```erlang
+{mod_pubsub, [{plugins, [<<"push">>]}]}, % mandatory minimal config
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults} % optional
+        ]}
+    ]}
+]}
+```
+
+#### Advantages
+* Completely XEP-0357 compliant
+* Most likely compatible with most 3rd party client libraries
+
+#### Drawbacks
+* Less efficient (PubSub has a considerable impact on heavily loaded systems)
+* More load within a single node
+* Harder to customise
+
+### MongooseIM as a PubSub-less XMPP server
+
+ ![](PushNotifications-no_pubsub.png)
+
+If compatibility with client libraries is not an issue, this is a really good approach.
+PubSub is completely bypassed and clients don't need to create a push node — if they attempt to
+do so, the server would respond with an error stanza, hence the incompatibility with [XEP-0357][].
+They only have to provide the virtual PubSub address in the enable stanza, and node name can be
+anything unique. In order to ensure uniqueness the APNS/FCM token can be used. Note that the token
+must be provided [as a publish options](https://xmpp.org/extensions/xep-0357.html#example-9) anyway.
+
+A direct connection to a push service (e.g. MongoosePush) must be configured on the same MongooseIM
+node, checkout [this][WithMongoosePush] tutorial on how to setup [MongoosePush][].
+
+```Erlang
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional
+            {virtual_pubsub_hosts, ["virtual.@HOST@"]}
+        ]}
+    ]}
+]}
+```
+
+#### Advantages
+* No need to use PubSub at all
+* More efficient (PubSub has a considerable impact on heavily loaded systems)
+* Simpler client-side usage
+
+#### Drawbacks
+* It is only partially XEP-0357 compliant, so 3rd party clients most probably won't work even in
```

Comment:
that's not true, XEP doesn't declare creation of the node as mandatory step. it declares only that PushService is represented by the `push PubSub node`. see [Business Rules](https://xmpp.org/extensions/xep-0357.html#sect-idm46544864749808) and [Enabling Notifications](https://xmpp.org/extensions/xep-0357.html#enabling) paragraphs

```
The App Server allocates or reuses a node on the App Server's XMPP Push Service
```

Reply:
also according to the ASCII diagram in XEP, it doesn't have to be done via XMPP client.